From 8b65999ff0c0e9e4e78f6cb2bef39208eda89004 Mon Sep 17 00:00:00 2001
From: Tommy Huang <tommy_huang@aspeedtech.com>
Date: Mon, 30 Mar 2026 17:46:24 +0800
Subject: [PATCH] drivers: reset-ast2700: fix the reset_status typo

Add register offset to the reset id over 32.

Signed-off-by: Tommy Huang <tommy_huang@aspeedtech.com>
Change-Id: I8faf229edb2205aeda5e6c94a27994be06a0d045
(cherry picked from commit 4792af600c5dd7fa2a4d55c52a76e6cb948122f0)

Signed-off-by: santhoshs <santhoshs@ami.com>
---
 drivers/reset/reset-ast2700.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/reset/reset-ast2700.c b/drivers/reset/reset-ast2700.c
index c2c27d36fb..abc7b709a0 100644
--- a/drivers/reset/reset-ast2700.c
+++ b/drivers/reset/reset-ast2700.c
@@ -45,7 +45,7 @@ static int ast2700_reset_status(struct reset_ctl *reset_ctl)
 	if (reset_ctl->id < 32)
 		status = BIT(reset_ctl->id) & readl(priv->base);
 	else
-		status = BIT(reset_ctl->id - 32) & readl(priv->base);
+		status = BIT(reset_ctl->id - 32) & readl(priv->base + 0x20);
 
 	return !!status;
 }
-- 
2.34.1

